include/tensor.tcc

```diff
@@ -113,140 +136,154 @@ void* Tensor<T>::data_view(const MemoryLayout mem_layout)
     */
 
-    void* ptr = 0;
+    void* ptr = NULL;
 
-    switch(mem_layout) {
-        case(MemoryLayout::contiguous) :
-            ptr = this->_data;
+    switch (mem_layout) {
+        case MemoryLayout::contiguous:
+            ptr = _data;
             break;
-        case(MemoryLayout::fortran_contiguous) :
-            ptr = this->_f_mem_views.allocate_bytes(
-                                     this->_n_data_bytes());
-            this->_c_to_f_memcpy((T*)ptr, (T*)this->_data, this->_dims);
+        case MemoryLayout::fortran_contiguous:
+            ptr = _f_mem_views.allocate_bytes(
+                                     _n_data_bytes());
+            _c_to_f_memcpy((T*)ptr, (T*)_data, _dims);
             break;
-        case(MemoryLayout::nested) :
-            this->_build_nested_memory(&ptr,
-                                       this->_dims.data(),
-                                       this->_dims.size(),
-                                       (T*)this->_data);
+        case MemoryLayout::nested:
+            _build_nested_memory(&ptr,
+                                       _dims.data(),
+                                       _dims.size(),
+                                       (T*)_data);
             break;
-        default :
+        default:
             throw std::runtime_error(
-                "Unsupported MemoryLayout "\
-                "value in "\
+                "Unsupported MemoryLayout value in "\
                 "Tensor<T>.data_view().");
     }
     return ptr;
 }
 
+// Fill a user provided memory space with values from tensor data
 template <class T>
 void Tensor<T>::fill_mem_space(void* data,
                                std::vector<size_t> dims,
                                MemoryLayout mem_layout)
 {
-    if(!this->_data)
+    if (_data == NULL) {
         throw std::runtime_error("The tensor does not have "\
                                  "a data array to fill with.");
+    }
 
-    if(dims.size() == 0)
-        throw std::runtime_error("The dimensions must have "\
-                                 "size greater than 0.");
+    if (dims.size() == 0) {
+        throw std::runtime_error("The dimensions must have nonzero size");
+    }
 
+    // Calculate size of memory buffer
     size_t n_values = 1;
     std::vector<size_t>::const_iterator it = dims.cbegin();
-    std::vector<size_t>::const_iterator it_end = dims.cend();
-    while(it!=it_end) {
-        if((*it)<=0)
-            throw std::runtime_error("All dimensions must "\
-                                     "be greater than 0.");
-        n_values*=(*it);
-        it++;
+    for ( ; it != dims.cend(); it++) {
+        if (*it <= 0) {
+            throw std::runtime_error("All dimensions must be greater than 0.");
+        }
+        n_values *= (*it);
     }
 
-    if(n_values!=this->num_values())
+    // Make sure there is space for all the data
+    // ***WS*** TBD: we should be checking each dimension here as well
```

Comment:
We should resolve this comment before merging.

Reply:
agree